py/src/braintrust/logger.py

```diff
@@ -524,11 +526,42 @@ def _check_json_serializable(event):
         raise Exception(f"All logged values must be JSON-serializable: {event}") from e
 
 
+class _BackgroundLogger(ABC):
+    @abstractmethod
+    def log(self, *args: LazyValue[Dict[str, Any]]) -> None:
+        pass
+
+    @abstractmethod
+    def flush(self, batch_size: Optional[int] = None):
+        pass
+
+
+class _MemoryBackgroundLogger(_BackgroundLogger):
```

Comment:
maybe we should move this out of logger.py into some testonly library?

Reply:
I left it in there becuase there is an already a testing one in there.